### PR TITLE
SoftMin Operator

### PR DESCRIPTION
**Pull request by haojin2**
## Description ##
Support softmin function: https://pytorch.org/docs/master/_modules/torch/nn/functional.html#softmin

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Softmin operator
- [x] Corresponding unit tests

## Comments ##
Passed more than 10000 trials on both CPU & GPU:
```
MXNET_TEST_COUNT=10000 nosetests -s --verbose tests/python/unittest/test_operator.py:test_softmin
[INFO] Setting module np/mx/python random seeds, use MXNET_MODULE_SEED=53121875 to reproduce.
test_operator.test_softmin ... ok

----------------------------------------------------------------------
Ran 1 test in 670.955s

OK
```
```
MXNET_TEST_COUNT=10000 nosetests -s --verbose tests/python/gpu/test_operator_gpu.py:test_softmin
[INFO] Setting module np/mx/python random seeds, use MXNET_MODULE_SEED=207344053 to reproduce.
test_operator_gpu.test_softmin ... ok

----------------------------------------------------------------------
Ran 1 test in 946.723s

OK
```

### CODE REVIEW

**Comment by haojin2** [comment]
The original softmax was not affected, tested with 10000 trials on both CPU & GPU:
```
MXNET_TEST_COUNT=10000 nosetests -s --verbose tests/python/unittest/test_operator.py:test_new_softmax
[INFO] Setting module np/mx/python random seeds, use MXNET_MODULE_SEED=2100317665 to reproduce.
test_operator.test_new_softmax ... ok

----------------------------------------------------------------------
Ran 1 test in 669.569s

OK
```
```
MXNET_TEST_COUNT=10000 nosetests -s --verbose tests/python/gpu/test_operator_gpu.py:test_new_softmax
[INFO] Setting module np/mx/python random seeds, use MXNET_MODULE_SEED=975663239 to reproduce.
test_operator_gpu.test_new_softmax ... ok

----------------------------------------------------------------------
Ran 1 test in 900.629s

OK
```

File: src/operator/nn/softmax.cc [hunk 1]
```diff
@@ -116,6 +116,45 @@ MXNET_OPERATOR_REGISTER_BINARY(_backward_softmax)
...
                                                         mxnet_op::softmax_bwd>);
 
+MXNET_OPERATOR_REGISTER_UNARY(softmin)
+.describe(R"code(Applies the softmin function.
+
+The resulting array contains elements in the range (0,1) and the elements along the given axis sum
+up to 1. Equivalant to `softmax(-z)` where `softmax(z)` is defined as:
+
+.. math::
+   softmax(\mathbf{z/t})_j = \frac{e^{z_j/t}}{\sum_{k=1}^K e^{z_k/t}}
```
Comment (szha) [inline src/operator/nn/softmax.cc]: formula needs correction

Comment (haojin2) [reply src/operator/nn/softmax.cc]: ok

File: src/operator/nn/softmax.cc [hunk 2]
```diff
@@ -116,6 +116,45 @@ MXNET_OPERATOR_REGISTER_BINARY(_backward_softmax)
...
+for :math:`j = 1, ..., K`
+
+t is the temperature parameter in softmax function. By default, t equals 1.0
+
+Example::
+
+  x = [[ 1.  2.  3.]
+       [ 3.  2.  1.]]
+
+  softmax(x,axis=0) = [[ 0.88079703,  0.5,  0.11920292],
```
[the remark this answers is not in the document]
Comment (haojin2) [reply src/operator/nn/softmax.cc]: Please read the doc carefully before you make a comment.

File: tests/python/unittest/test_operator.py [hunk 3]
```diff
@@ -4519,6 +4519,23 @@ def test_invalid_shape():
     test_where_numeric_gradient((5, 7, 9), False)
     test_invalid_shape()
 
+
+@with_seed()
+def test_softmin():
+    for ndim in range(1, 5):
+        shape = np.random.randint(1, 5, size=ndim)
```
[the remark this answers is not in the document]
Comment (haojin2) [reply tests/python/unittest/test_operator.py]: I can definitely add that

[the remark this answers is not in the document]
Comment (haojin2) [reply tests/python/unittest/test_operator.py]: I just realized that checks for different dtypes are lacking in all of softmax-related tests, I'll add those checks for softmin only in this PR and submit another PR for adding those checks for other tests.

File: src/operator/nn/softmax.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (anirudhacharya) [reply src/operator/nn/softmax.cc]: resolved offline to change it to softmin.

**Comment by haojin2** [comment]
@szha
